specification/elasticsan/resource-manager/Microsoft.ElasticSan/stable/2023-01-01/elasticsan.json

```diff
@@ -0,0 +1,2747 @@
+            "create"
+          ]
+        },
+        "totalVolumeSizeGiB": {
+          "description": "Total size of the provisioned Volumes in GiB.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "volumeGroupCount": {
+          "description": "Total number of volume groups in this Elastic San appliance.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "totalIops": {
+          "description": "Total Provisioned IOPS of the Elastic San appliance.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "totalMBps": {
+          "description": "Total Provisioned MBps Elastic San appliance.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "totalSizeTiB": {
+          "description": "Total size of the Elastic San appliance in TB.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "privateEndpointConnections": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          },
+          "description": "The list of Private Endpoint Connections."
+        },
+        "publicNetworkAccess": {
+          "type": "string",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "PublicNetworkAccess",
+            "modelAsString": true
+          },
+          "description": "Allow or disallow public network access to ElasticSan. Value is optional but if passed in, must be 'Enabled' or 'Disabled'."
+        }
+      }
+    },
+    "ElasticSanUpdate": {
+      "type": "object",
+      "description": "Response for ElasticSan update request.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of ElasticSan.",
+          "$ref": "#/definitions/ElasticSanUpdateProperties"
+        },
+        "tags": {
+          "type": "object",
+          "description": "Update tags",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ElasticSanUpdateProperties": {
+      "type": "object",
+      "description": "Elastic San update properties.",
+      "properties": {
+        "baseSizeTiB": {
+          "description": "Base size of the Elastic San appliance in TiB.",
+          "type": "integer",
+          "format": "int64",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "extendedCapacitySizeTiB": {
+          "description": "Extended size of the Elastic San appliance in TiB.",
+          "type": "integer",
+          "format": "int64",
+          "x-ms-mutability": [
+            "read",
+            "create"
```

Comment:
For update, the values of x-ms-mutability should include "update", (but might not create or read)